src/esp/assets/ResourceManager.h

```diff
@@ -1108,6 +1128,16 @@ class ResourceManager {
    * @brief See @ref setRecorder.
    */
   std::shared_ptr<esp::gfx::replay::Recorder> gfxReplayRecorder_;
+
+  /**
+   * @brief The imaged based lighting for PBR, each is a collection of
+   * an environment map, an irradiance map, a BRDF lookup table (2D texture),
+   * and a pre-fitered map
+   */
+  std::map<int, std::unique_ptr<esp::gfx::PbrImageBasedLighting>>
```

Comment:
Shouldn't this be an std::unordered_map? Is the order important?

Reply:
It has nothing to do with the order. Also It will not make too much difference in terms of performance. 

`unordered_map` is implemented in a linked list and thus extremely not cache-friendly. So even though the time complexity in theory is O(1) for finding and inserting an element, the overhead is actually very high. Avoiding `unordered_map` or `unordered_set` is almost a common sense. I rarely use them.

`map` has insertion time `O(logN)`. However in our usage case, perhaps we will never ever require, say 10 of such an element in the future, so you can safely assume the time cost is approximately constant.

Reply:
> is implemented in a linked list and thus extremely not cache-friendly

That's only partially correct. Neither `map` nor `unordered_map` are cache friendly, both are extremely awful _by design_ as each element has to be a separate allocation in order to satisfy the iterator invalidation requirements. The linked list only gets used when more than element falls into the same bucket, but with ~10 elements it's unlikely.

> Avoiding `unordered_map` or `unordered_set` is almost a common sense.

I'd say otherwise -- the memory-jumping overhead of balancing a tree for `map` insertion (or memory jumps for doing the binary search) is _bigger_ than basically a single lookup into a hashmap bucket. The "common sense" is to avoid any STL maps, be it a tree or a hashmap, but when you have no better option (because I didn't get to implementing better alternatives in Magnum yet), an `unordered_map` is less evil than a `map` ;)

Reply:
Good to know something new. Thank you, @mosra .

Considering the problem scale, I do not think there is too much difference between `map` and `unordered_map`. (We only have 1 environment map which may stay for quite a long time unless we would like to have different environment map for different rooms. e.g.). 

So I will just keep the `map` here.

Reply:
You know what, I changed it to `vector` given the fact that the environment maps are static and will not be dynamically created or destroyed during the runtime (unlike the dynamic object). So I am now using the `vector` here.